### PR TITLE
feat(TFD-1696) Add new converters for Streams TypeConverter

### PR DESCRIPTION
**Pull request by lbourgeois**
**What is the problem this Pull Request is trying to solve?**
A new component is being developed for DataStreams
This new component is a TypeConverter which has to handle (among others) Date/DateTime/BigDecimal types

**What is the chosen solution to this problem?**
Using Daikon converters after having done needed improvements

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TFD-1696
 
**Please check if the Pull Request fulfills these requirements**
- [X] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [X] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/278/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/278/)

File: daikon/src/main/java/org/talend/daikon/converter/BigDecimalConverter.java [hunk 1]
```diff
@@ -12,8 +16,10 @@
 
     public static String ROUNDING_MODE = "roundingMode";
 
+    public static String DECIMAL_FORMAT = "decimalFormat";
+
     @Override
-    public BigDecimal convert(Object value) {
+    public BigDecimal convert(Object value) throws TalendRuntimeException {
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/converter/BigDecimalConverter.java]: no need to throw a RuntimeException

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/279/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/279/)

File: daikon/src/main/java/org/talend/daikon/converter/BigDecimalConverter.java (hunk 1 above)
Comment (lbourgeois) [reply daikon/src/main/java/org/talend/daikon/converter/BigDecimalConverter.java]: It should throw the original ParseException ?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/280/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/280/)

**Comment by lbourgeois** [comment]
@sgandon I followed your requested changes, can you review please ?

**Pull request merged**